xdsl/dialects/gpu.py

```diff
@@ -671,6 +676,22 @@ def __init__(self):
         super().__init__(result_types=[IndexType()])
 
 
+@irdl_op_definition
+class PrintfOp(IRDLOperation):
+    name = "gpu.printf"
+    format: StringAttr = attr_def(StringAttr)
```

Comment:
```suggestion
    format: StringAttr = prop_def(StringAttr)
```
This is what is mismatching with the MLIR implementation :slightly_smiling_face: (You'll need to update the import accordingly!)

MLIR enables properties at the Dialect level IIRC, we define individual attributes as properties like this. This might change soon, as this is unnecessary and confusing!